libcaf_net/caf/net/http/server_factory.hpp

```diff
@@ -73,16 +73,19 @@ class http_conn_factory
   using connection_handle = typename Transport::connection_handle;
 
   http_conn_factory(std::vector<net::http::route_ptr> routes,
-                    size_t max_consecutive_reads)
+                    size_t max_consecutive_reads, size_t max_request_size)
     : routes_(std::move(routes)),
-      max_consecutive_reads_(max_consecutive_reads) {
+      max_consecutive_reads_(max_consecutive_reads),
+      max_request_size_(max_request_size) {
     // nop
   }
 
   net::socket_manager_ptr make(net::multiplexer* mpx,
                                connection_handle conn) override {
     auto app = net::http::router::make(routes_);
     auto serv = net::http::server::make(std::move(app));
+    if (max_request_size_ > 0)
+      serv->max_request_size(max_request_size_);
```

Comment:
I would pass the argument through without checking and instead do the checking in the server, since the server should sanity-check/verify its arguments.